homeassistant/components/livisi/coordinator.py

```diff
@@ -70,67 +63,66 @@ async def async_setup(self) -> None:
             await self.aiolivisi.async_set_token(
                 livisi_connection_data=livisi_connection_data
             )
-        controller_info = await self.aiolivisi.async_get_controller()
-
-        controller_data = controller_info
+        controller_data = await self.aiolivisi.async_get_controller()
         if controller_data.get("controllerType") == "Avatar":
             self.port = AVATAR_PORT
             self.is_avatar = True
         else:
             self.port = CLASSIC_PORT
             self.is_avatar = False
-        self.serial_number = controller_data.get("serialNumber")
-        self.controller_type = controller_data.get("controllerType")
-        self.os_version = controller_data.get("osVersion")
+        self.serial_number = controller_data["serialNumber"]
+        self.controller_type = controller_data["controllerType"]
+        self.os_version = controller_data["osVersion"]
 
-    async def async_get_devices(self) -> list:
+    async def async_get_devices(self) -> list[dict[str, Any]]:
         """Set the discovered devices list."""
-        shc_devices = await self.aiolivisi.async_get_devices()
-        if bool(shc_devices) is False:
-            shc_devices = await self.aiolivisi.async_get_devices()
-        return shc_devices
+        return await self.aiolivisi.async_get_devices()
 
-    async def async_get_pss_state(self, capability):
+    async def async_get_pss_state(self, capability: str) -> bool:
         """Set the PSS state."""
-        response = await self.aiolivisi.async_get_pss_state(capability[1:])
+        response: dict[str, Any] = await self.aiolivisi.async_get_pss_state(
+            capability[1:]
+        )
         if response is None:
-            return
-        on_state = response.get("onState")
-        return on_state.get("value")
+            return None
```

Comment:
Either return `False` or update the return value type annotation.

Reply:
I changed the return value type annotation to accept bool or None.